### ISSUE TITLE
Full-animated cursor not always animated

### ISSUE THREAD

**Issue by scott092707** [swaywm/sway#6891]
### Please fill out the following:
$ apt-cache policy sway
sway:
  Installed: 1.6.1-2

$ apt-cache policy *wlroots*
...
libwlroots9: 
  Installed: 0.14.1-5                                                                                                                                                     ...
...

$ uname -a
Linux ASUS-Prime-B350MA 5.15.0-2-amd64 #1 SMP Debian 5.15.5-2 (2021-12-18) x86_64 GNU/Linux
                                                                                                                                                    
cat /etc/debian_version
bookworm/sid

https://gist.github.com/scott092707/678027886911af118499a31bf943291f
contains:
- **Debug Log:**
- **Configuration File:**
- sway starting script and environment (pulled in from start script)
- one or two other possibly useful things

- **Stack Trace:**
  - This is only needed if sway crashes.
  - [did not crash]

- **Description:**
  - The steps you took in plain English to reproduce the problem.
  - 
I am not sure just how to characterize this issue, except for
describing what happens - where it happens, and where it does not.

I use a 49 in. 4K TV as a computer monitor, and even though I have installed 
a red cursor - oxy-red-argentina, and use it at its 48px size, I still sometimes
lose track of where it is.

I got the idea to animate all (or at least the most commonly used) cursor glyphs with
color only.  The shape stays the same, just the color animates - cycling between
three colors: the original red, and black and white (easy to edit, with GIMP's
Colors->Threshold tool: send the slider all the way to one side for black, to
the other for white).  It's hard to imagine a situation where a cursor changing
from red-to-black-to-white would still be hard to find.
Also, since 48px is fairly large, and I do NOT scale my output, the cursor
sometimes covers up something I want to see.  So, while editing the cursors,
I also hollowed them out, leaving mainly the outline.

GIMP is able to read and write X11 Mouse Cursors.  I just need to modify the 
layers' name to indicate animation, and remove the .xmc extension afterwards.

This works.

Not everywhere, though...

[I'm now going to substitute the term "blink" for "animate", as being shorter.]

Cursors in Sway's workspace background, swaynag, swaybar and window titlebars/borders
do NOT blink.

Mate's file manager Caja, which runs under Xwayland, does NOT blink.
GIMP and Hugin, likewise Xwayland, do not blink.
Aha!, I hear you say: it's an Xwayland vs. Wayland problem.
No.
1. If I run Firefox under Xwayland, the cursor blinks, as it does under Wayland.
2. Even if in GIMP/etc. the standard "left_ptr", "pointing_hand", etc.
do not blink, if there is a text input box (such as GIMP's Zoom box, or editing a
layer name) or interior separator line (such as between Caja's sidebar and the
main file/directory icon section), those cursors ("xterm", "col-resize",
"[e-][w-]resize") DO blink. 

Cursors in all Qt programs that I have tried (all of which run natively under Wayland)
DO blink.  These include Qterminal, featherpad, lximage-qt, QtPass, speedcrunch, ...
But it is apparently not a Qt vs Gtk+ issue, as the cursor blinks under Atril and eom,
which are also from Mate, like Caja, which does not.

When the cursor does not blink, it shows just the red layer of the given cursor glyph.
This IS my altered cursor, not some other non-edited cursor, because it is hollowed
out, where the original cursor is not.

**Comment by emersion** [comment]
Sway, and some clients like swaybar, don't support animated cursors at all. Converting this into a feature request.

**Comment by scott092707** [comment]
I'm not sure if it was an intentional fix, or something else that inadvertently fixed this too, but anything sway related
("Sway's workspace background, swaynag, swaybar and window titlebars/borders") now DO work.  I don't know when
exactly it got fixed (1.8?), but I am grateful.  Thank you!
I believe that the issue for other apps, as described above, still applies.  I know it does for Caja...

**Comment by Nefsen402** [comment]
Ah, this has been fixed for a while now. https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4117 If there is a circumstance where animated cursors don't work, that's up to the application developer.
